src/amdinfer/workers/rocal.cpp

```diff
@@ -0,0 +1,438 @@
+#include "amdinfer/core/data_types.hpp"         // for DataType, operator<<
+#include "amdinfer/core/exceptions.hpp"         // for invalid_argument, runt...
+#include "amdinfer/core/inference_request.hpp"  // for InferenceRequest
+#include "amdinfer/core/inference_response.hpp"  // for InferenceResponse
+#include "amdinfer/core/parameters.hpp"          // for ParameterMap
+#include "amdinfer/declarations.hpp"             // for InferenceResponseOutput
+#include "amdinfer/observation/logging.hpp"  // for AMDINFER_LOG_INFO, AMD...
+#include "amdinfer/observation/metrics.hpp"  // for Metrics, MetricCounterIDs
+#include "amdinfer/util/containers.hpp"      // for containerProduct
+#include "amdinfer/util/memory.hpp"          // for copy
+#include "amdinfer/util/queue.hpp"           // for BufferPtrsQueue
+#include "amdinfer/util/string.hpp"          // for contains
+#include "amdinfer/util/thread.hpp"          // for setThreadName
+#include "amdinfer/util/timer.hpp"           // for Timer
+#include "amdinfer/workers/worker.hpp"       // for Worker, kNumBufferAuto
+#include "amdinfer/buffers/vector.hpp"       // for VectorBuffer
+
+namespace amdinfer::workers {
+
+/**
+ * @brief Rocal Operation Enum Class
+ */
+enum class RocalOperation {
+    Resize, ResizeMirrorNormalize, CropResizeFixed, 
+    CropMirrorNormalize, CropFixed, ResizeCropMirrorFixed
+};
+
+/**
+ * @brief A helper function to convert string to RocalOperation enum
+ */
+RocalOperation getOperationType(const std::string& op) {
+    if (op == "Resize") return RocalOperation::Resize;
+    if (op == "ResizeMirrorNormalize") return RocalOperation::ResizeMirrorNormalize;
+    if (op == "CropResizeFixed") return RocalOperation::CropResizeFixed;
+    if (op == "CropMirrorNormalize") return RocalOperation::CropMirrorNormalize;
+    if (op == "CropFixed") return RocalOperation::CropFixed;
+    if (op == "ResizeCropMirrorFixed") return RocalOperation::ResizeCropMirrorFixed;
+    std::cerr << "Unknown operation: " + op << std::endl;
+    throw std::invalid_argument("Unknown operation: " + op);
+}
+
+/**
+ * @brief The rocAL worker class
+ */
+class RocalWorker : public SingleThreadedWorker {
+ public:
+  using SingleThreadedWorker::SingleThreadedWorker;
+  [[nodiscard]] std::vector<MemoryAllocators> getAllocators() const override;
+
+ private:
+  void doInit(ParameterMap* parameters) override;
+  void doAcquire(ParameterMap* parameters) override;
+  BatchPtr doRun(Batch* batch, const MemoryPool* pool) override;
+  void doRelease() override;
+  void doDestroy() override;
+
+  void deserialize(const std::string& model_path);
+  std::vector<float> parseVector(const rapidjson::Value& jsonArray);
+
+  std::string model_path_;
+
+  // The mode for rocalExternalSource reader
+  // 0 - External File Mode
+  // 1 - Raw Compressed Mode
+  // 2 - Raw Uncompressed Mode
+  const int rocal_external_mode_ = 1;
+
+  // Decode width & height
+  const int decode_width_ = 1000, decode_height_ = 1000; 
```

Comment:
rocAL loader requires a max decode_width / height value to initialize the handle. 
Since the image size is unknown until the runtime, we give a large value initially and resize the image to given width / height.
I'll rename it to `max_decode_width / max_decode_height`

Reply:
Yes, that's better thanks